pkg/webui/console/containers/fetch-select/index.js

```diff
@@ -0,0 +1,97 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { connect as storeConnect } from 'react-redux'
+
+import PropTypes from '../../../lib/prop-types'
+import Field from '../../../components/form/field'
+import Select from '../../../components/select'
+
+const formatOptions = formats =>
+  Object.keys(formats).map(key => ({ value: key, label: formats[key] }))
+
+export default function({
+  optionsSelector,
+  errorSelector,
+  fetchingSelector,
+  fetchOptions,
+  warningMessage,
+  titleMessage,
```

Comment:
With this implementation, there is no way to enforce `titleMessage` to be present, except for checking it in the ui. Would be nice to add prop-types validation here to see errors.

Reply:
I don't know if the prop-types pattern should be applied in this particular case because the component is built from a generator function, where I simply have to assume the arguments have been passed in correctly (like we do in other places too).
What can already help is to set the `title` prop of the field component as required.
Let me know if you have any other idea in mind.